src/dvb.c

```diff
@@ -412,38 +436,117 @@ void dvb_set_demux_source(adapter *ad)
 		if (dmx >= 0)
 			close(dmx);
 	}
+#elif defined(GXAPI)
+#define TS_DEMUX_IN_GATE  (188 * 200)
+	GxDemuxProperty_ConfigDemux demux = { 0 };
+
+	LOG("GXAPI MUXTS: Open DVB device - TS source %d, ts select %d, stream mode %d\n", (ad->gx_ts_config >> 2) & 0x03, (ad->gx_ts_config >> 1) & 0x01, ad->gx_ts_config & 0x01);
+
+	demux.source = (ad->gx_ts_config >> 2) & 0x03; /* DEMUX_TS1 */
+	demux.ts_select = (ad->gx_ts_config >> 1) & 0x01; /* FRONTEND */
+	demux.stream_mode = ad->gx_ts_config & 0x01; /* DEMUX_PARALLEL */
+	demux.time_gate = 0xf;
+	demux.byt_cnt_err_gate = 0x03;
+	demux.sync_loss_gate = 0x03;
+	demux.sync_lock_gate = 0x03;
+
+	ad->ret_prop = GxAVSetProperty(ad->dvr, ad->module, GxDemuxPropertyID_Config, &demux, sizeof(GxDemuxProperty_ConfigDemux));
+	if(ad->ret_prop < 0)
+	{
+		LOG("GXAPI: [%s] Not set property....", __FUNCTION__);
+		GxAVCloseModule(ad->dvr, ad->module);
+		GxAVDestroyDevice(ad->dvr);
+		return;
+	}
+
+	/* special slot */
+	ad->muxslot.type = DEMUX_SLOT_MUXTS;
+	ad->muxslot.flags = (DMX_TSOUT_EN | DMX_CRC_DISABLE | DMX_REPEAT_MODE);
+	ad->ret_prop = GxAVGetProperty(ad->dvr, ad->module, GxDemuxPropertyID_SlotAlloc, &ad->muxslot, sizeof(GxDemuxProperty_Slot));
+	if (ad->ret_prop < 0)
+	{
+		LOG("GXAPI MUXTS: GxDemuxPropertyID_SlotAlloc Problem...");
+		GxAVCloseModule(ad->dvr, ad->module);
+		GxAVDestroyDevice(ad->dvr);
+		return;
+	}
+
+	/* special filter */
+	ad->muxfilter.slot_id = ad->muxslot.slot_id;
+	ad->muxfilter.sw_buffer_size = DVR_BUFFER;
+	ad->muxfilter.almost_full_gate = TS_DEMUX_IN_GATE;
+	ad->ret_prop = GxAVGetProperty(ad->dvr, ad->module, GxDemuxPropertyID_FilterAlloc, &ad->muxfilter, sizeof(GxDemuxProperty_Filter));
+	if (ad->ret_prop < 0)
+	{
+		LOG("GXAPI TS Filter: GxDemuxPropertyID_FilterAlloc Problem...");
+		GxAVCloseModule(ad->dvr, ad->module);
+		GxAVDestroyDevice(ad->dvr);
+		return;
+	}
+	ad->demux_lock = 1;
 #endif
 }
 
 int dvb_open_device(adapter *ad)
 {
 	char buf[100];
+	int fn = ad->fn;
```

Comment:
@McMCCRU , why have you defined a new variable `fn` here? Is there any reason to not use `ad->fn` ?

Reply:
Hi! 
This was done to support combo devices with using an optional external demodulator and tuner(DVB-C/T/T2) on the gx6605s. The DVB GXAPI is based on the old Linux DVB API v.3.x and does not support the operation of a multi-frontend with one demultiplexer, so I made it so that would not write set values fn to ad->fn, which is further used in original minisatip.
P.S. If I'm not mistaken, then the minisatip itself does not support multi frontend on one adapter, only one adapter and one frontend...